pandas/tseries/tdi.py

```diff
@@ -440,21 +439,18 @@ def astype(self, dtype):
 
         if dtype == np.object_:
             return self.asobject
-        elif dtype == _INT64_DTYPE:
-            return self.asi8.copy()
         elif dtype == _TD_DTYPE:
```

Comment:
In `pandas/core/common.py`:
```python
def is_timedelta64_dtype(arr_or_dtype):
    tipo = _get_dtype_type(arr_or_dtype)
    return issubclass(tipo, np.timedelta64)

def is_timedelta64_ns_dtype(arr_or_dtype):
    tipo = _get_dtype_type(arr_or_dtype)
    return tipo == _TD_DTYPE
```
The first function is ok. In the other, it should be `tipo = _get_dtype(arr_or_dtype)`. Otherwise it always returns `False`. (It was probably a copy and paste from above). 

I fixed it and it's just passed local tests.